problem-spring-webflux/src/main/java/org/zalando/problem/spring/webflux/advice/security/SecurityProblemSupport.java

```diff
@@ -3,40 +3,94 @@
+import java.lang.reflect.Method;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
 import static org.apiguardian.api.API.Status.STABLE;
 
 @API(status = STABLE)
 @Component
 public class SecurityProblemSupport implements ServerAuthenticationEntryPoint, ServerAccessDeniedHandler {
 
-    private final SecurityAdviceTrait advice;
     private final ObjectMapper mapper;
 
+    private final Map<ControllerAdviceBean, ExceptionHandlerMethodResolver> exceptionHandlerAdviceCache =
+            new LinkedHashMap<>(64);
+
     @Autowired
-    public SecurityProblemSupport(final SecurityAdviceTrait advice, final ObjectMapper mapper) {
-        this.advice = advice;
+    public SecurityProblemSupport(final ApplicationContext ctx, final ObjectMapper mapper) {
         this.mapper = mapper;
+        initControllerAdviceCache(ctx);
     }
 
     @Override
     public Mono<Void> commence(final ServerWebExchange exchange, final AuthenticationException e) {
-        return advice.handleAuthentication(e, exchange)
-                .flatMap(entity -> AdviceUtils.setHttpResponse(entity, exchange, mapper));
+        return resolveException(exchange, e);
     }
 
     @Override
     public Mono<Void> handle(final ServerWebExchange exchange, final AccessDeniedException e) {
-        return advice.handleAccessDenied(e, exchange)
-                .flatMap(entity -> AdviceUtils.setHttpResponse(entity, exchange, mapper));
+        return resolveException(exchange, e);
+    }
+
+    private void initControllerAdviceCache(ApplicationContext applicationContext) {
+        List<ControllerAdviceBean> beans = ControllerAdviceBean.findAnnotatedBeans(applicationContext);
+        AnnotationAwareOrderComparator.sort(beans);
+
+        for (ControllerAdviceBean bean : beans) {
+            Class<?> beanType = bean.getBeanType();
+            if (beanType != null) {
+                ExceptionHandlerMethodResolver resolver = new ExceptionHandlerMethodResolver(beanType);
+                if (resolver.hasExceptionMappings()) {
+                    this.exceptionHandlerAdviceCache.put(bean, resolver);
+                }
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private Mono<Void> resolveException(ServerWebExchange exchange, Exception e) {
+        Object targetBean = null;
+        Method targetMethod = null;
+        for (ControllerAdviceBean advice : this.exceptionHandlerAdviceCache.keySet()) {
+            targetBean = advice.resolveBean();
+            targetMethod = this.exceptionHandlerAdviceCache.get(advice).resolveMethodByThrowable(e);
+            if (targetMethod != null) {
+                break;
+            }
+        }
+        if (targetMethod == null) {
+            return null;
+        }
+
+        try {
+            ReflectionUtils.makeAccessible(targetMethod);
+            Mono<ResponseEntity<Problem>> o = (Mono<ResponseEntity<Problem>>) targetMethod.invoke(targetBean, e, exchange);
```

Comment:
This is rather limiting compared to what users are familiar with in regards to `@ExceptionHandler` methods, isn't it?